### PR TITLE
Add flags and set maximum arguments for all client side commands #83 #84

### LINKED ISSUE #83: bug: Create flags for commands instead of parsing them from cli
Description:

Flags provide an intuitive way of passing command line arguments and it will be good if we can provide flags for commands like `getstream`, `describe` and `execute`. Presently we take the arguments from cli and parse them.

Present Behaviour:

The string from cli is getting parsed

Expected Behaviour:

Flags should be used to pass in arguments

### LINKED ISSUE #84: bug: list command accepts arguments
#### Description
`list` command is built in a way to accept 1 argument. However, the basic definition of list needs 0 arguments, and giving argument to list command makes no sense as arguments are not utilised in any way.

#### Present Behaviour
`list` can accept any 1 arbitrary argument and still function correctly.

#### Expected Behaviour
`list` should throw an error if any arguments are given along with it.

### PR DESCRIPTION
**Pull request by darshak-sorathiya**
#### Description
This pull request implements flags for getstream and execute commands. Also set maximum arguments for all client side commands. fixes #83, fixes #84  

#### Present Behaviour
The execute command parses the arguments and no flags are implemented for it. Also, the maximum limit of arguments is not set for most of the commands.

#### Expected Behaviour 
As described above.

### CODE REVIEW

File: internal/cli/command/getstream/getstream.go [hunk 1]
```diff
@@ -12,18 +12,22 @@ import (
...
-	return &cobra.Command{
+
+	var jobID string
+
+	getStreamCmd := &cobra.Command{
 		Use:   "getstream",
 		Short: "Get job log data",
 		Long:  `Get job log by giving arguments`,
-		Args:  cobra.MinimumNArgs(1),
+		Args:  cobra.MaximumNArgs(0),
```
Comment (yashipro13) [inline internal/cli/command/getstream/getstream.go:22]: Exact Args will be better with this usage

File: internal/cli/command/create/create.go [hunk 2]
```diff
@@ -44,7 +45,7 @@ func NewCmd(octaviusDaemon daemon.Client, fileUtil file.File) *cobra.Command {
 		},
 	}
 	createCmd.Flags().StringVarP(&metadataFilePath, "job-path", "", "", "path to metadata.json(required)")
-	createCmd.MarkFlagRequired("job-path")
+	_ = createCmd.MarkFlagRequired("job-path")
```
Comment (littlestar642) [inline internal/cli/command/create/create.go:48]: should not leave the error unhandled.

File: internal/cli/command/config/config.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ func NewCmd(fileUtil file.File) *cobra.Command {
 		Short:   "Configure octavius client",
 		Long:    "This command helps configure client with control plane host, email id and access token",
 		Example: "octavius config [flags]",
+		Args:    cobra.MaximumNArgs(0),
```
Comment (littlestar642) [inline internal/cli/command/config/config.go:28]: Just see if ExactArgs will be a better choice in steaf of MaximumNArgs. You can use whatever is more semantic.

File: internal/cli/command/execution/execution.go [hunk 4]
```diff
@@ -40,4 +51,10 @@ func NewCmd(octaviusDaemon daemon.Client) *cobra.Command {
 			printer.Println(fmt.Sprintf("Job has been added to pending list successfully.\nYou can see the execution logs using getstream %s", response.Status), color.FgGreen)
 		},
 	}
+
+	executionCmd.Flags().StringVarP(&jobName, "job-name", "", "", "It contains Job Name")
+	executionCmd.MarkFlagRequired("job-name")
```
Comment (littlestar642) [inline internal/cli/command/execution/execution.go:56]: handle the errors returned. It will be good in making a robust application

**Review by littlestar642 — CHANGES_REQUESTED** [review]

File: internal/cli/command/create/create.go (hunk 2 above)
Comment (adbusa67) [reply internal/cli/command/create/create.go:48]: this is handled in #87

File: internal/cli/command/execution/execution.go (hunk 4 above)
Comment (adbusa67) [reply internal/cli/command/execution/execution.go:56]: this is handled in #87

**Comment by littlestar642** [comment]
closing since the changes are already integrated in #87

**Pull request closed without merging**